test/test.make

```diff
@@ -184,8 +184,22 @@ RUN_E2E = KUBECONFIG=`pwd`/_work/$(CLUSTER)/kube.config \
 		-ginkgo.randomizeAllSpecs=false \
 	        $(TEST_E2E_ARGS) \
                 -report-dir=$(TEST_E2E_REPORT_DIR)
+RUN_OPERATOR_E2E = KUBECONFIG=`pwd`/_work/$(CLUSTER)/kube.config \
+	REPO_ROOT=`pwd` \
+	CLUSTER=$(CLUSTER) \
+	$(shell source test/test-config.sh; \
+	  echo PMEM_CSI_IMAGE=$$TEST_LOCAL_REGISTRY/pmem-csi-driver:$(IMAGE_VERSION); \
+	) \
+	GO='$(GO)' \
+	$(GO) test -count=1 -timeout 0 -v ./test/e2e/operator \
+                -ginkgo.skip='$(subst $(space),|,$(strip $(subst @,$(space),$(TEST_E2E_SKIP_ALL))))' \
+                -ginkgo.focus='$(subst $(space),|,$(strip $(subst @,$(space),$(TEST_E2E_FOCUS))))' \
+		-ginkgo.randomizeAllSpecs=false \
+	        $(TEST_E2E_ARGS) \
+                -report-dir=$(TEST_E2E_REPORT_DIR)
 test_e2e: start $(RUN_TEST_DEPS)
 	$(RUN_E2E)
+	$(RUN_OPERATOR_E2E)
```

Comment:
Can you integrate this into the normal `test/e2e` suite? Then we don't need all of this duplicate code (`RUN_OPERATOR_E2E`, E2E boilerplate code in `operator_test.go`) and don't need to worry about globbering the content of `TEST_E2E_REPORT_DIR`.

Also, the tests must be non-destructive, i.e. it must be possible to run them repeatedly against the same test cluster. Not sure whether that is currently the case: when the operator deploys in LVM mode, will all available space be used? We have to use the same approach as for the deployment via yaml (= use half of the PMEM for LVM, leave the rest free) - https://github.com/intel/pmem-csi/blob/da7c9991944d55dd349385656497f7cd69b8429c/test/setup-deployment.sh#L136-L139

Reply:
Current deployment api does not support choosing `--use-of-dax` option. This was one reason  i chose to run the operator tests after running existing driver tests, so that the it won't pollute the cluster space. But, supporting this is a must in order to run driver tests against the driver deployed with the operator. I will add support for this as this needs  

As a workaround i found other solution for this by using fake image for deployments. Obviously, the current operator tests only test if all sub-resources(StatefulSet, DaemonSet, CSIDriver, Secrets, etc) and are initialized with appropriate values chosen by the deployment.

Reply:
> This was one reason i chose to run the operator tests after running existing driver tests, so that the it won't pollute the cluster space.

When you make design decisions like that, can you document this in the code with comments for your fellow developers?

Reply:
> When you make design decisions like that, can you document this in the code with comments for your fellow developers?

Ok will do. Anyways I reverted this and as you pointed now operator tests are part of same testsuties of other e2e tests.

Reply:
> Current deployment api does not support choosing `--use-of-dax` option. This was one reason i chose to run the operator tests after running existing driver tests, so that the it won't pollute the cluster space. But, supporting this is a must in order to run driver tests against the driver deployed with the operator. I will add support for this as this needs

@pohly I added support for configuring lvm driver space usage: https://github.com/intel/pmem-csi/pull/559/commits/b35896903f392ac60cc2c800237b92378f33207d

can you check if its ok to merge this PR.